components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java

```diff
@@ -288,70 +317,39 @@ public static SessionDataStore getInstance() {
         return instance;
     }
 
+    // Return Jedis object.
+    private static BinaryJedis getJedisInstance() {
+
+        if (jedis != null)
+            return jedis;
+        return new BinaryJedis(HOST, PORT);
+    }
```

Comment:
Can we guarantee that this new instance creation would be thread-safe?  Shall we extract this out and use the singleton pattern here? WDYT?

Reply:
yeah aiya. Using singleton is better here.